src/battle/battle_system.c

```diff
@@ -1303,3 +1305,122 @@ u32 CalcMoneyLoss(Party *party, PlayerProfile *profile) {
     
     return loss;
 }
+
+void BattleSystem_SetPokedexSeen(BattleSystem *bsys, int battlerId) {
+    u32 flag = ov12_02261258(bsys->opponentData[battlerId]);
+    Pokemon *mon = BattleSystem_GetPartyMon(bsys, battlerId, ov12_022581D4(bsys, bsys->ctx, 2, battlerId));
+
+    if (!(bsys->battleType & (BATTLE_TYPE_LINK | BATTLE_TYPE_TOWER))) {
+        if ((flag & 1) || bsys->battleType == (BATTLE_TYPE_6 | BATTLE_TYPE_DOUBLES | BATTLE_TYPE_MULTI) || bsys->battleType == (BATTLE_TYPE_TRAINER | BATTLE_TYPE_6 | BATTLE_TYPE_DOUBLES | BATTLE_TYPE_MULTI)) {
+            Pokedex_SetMonSeenFlag(bsys->pokedex, mon);
+        }
+    }
+    if (!(flag & 1) && GetMonData(mon, MON_DATA_SPECIES_OR_EGG, NULL) == SPECIES_BURMY) {
+        Pokedex_SetMonCaughtFlag(bsys->pokedex, mon);
+    }
+}
+
+void BattleSystem_SetPokedexCaught(BattleSystem *bsys, int battlerId) {
+    u32 flag = ov12_02261258(bsys->opponentData[battlerId]);
+    
+    if (!(bsys->battleType & (BATTLE_TYPE_LINK | BATTLE_TYPE_TOWER)) && (flag & 1)) {
+        int selectedMonIndex = ov12_022581D4(bsys, bsys->ctx, 2, battlerId);
+        Pokemon *mon = BattleSystem_GetPartyMon(bsys, battlerId, selectedMonIndex);
+        Pokedex_SetMonCaughtFlag(bsys->pokedex, mon);
+    }
+}
+
+BOOL BattleSystem_CheckMonCaught(BattleSystem *bsys, int battlerId) {
+    return Pokedex_CheckMonCaughtFlag(bsys->pokedex, battlerId);
+}
+
+void BattleSystem_SetDefaultBlend() {
+    G2_BlendNone();
+}
+
+u8 BattleSystem_PrintTrainerMessage(BattleSystem *bsys, int trainerId, int battlerId, int a2, int delay) {
+    Window *window = BattleSystem_GetWindow(bsys, 0);
+    int index;
+    
+    if (bsys->battleType & BATTLE_TYPE_TOWER) {
+        if (trainerId == 0x2710 || bsys->battleType & BATTLE_TYPE_13) {
```

Comment:
trainer ID constant? not sure if we already have these documented

Reply:
Can;t find it if we do